src/Qwiq.Core/IQueryFolder.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Microsoft.Qwiq
+{
+    public interface IQueryFolder : IIdentifiable<Guid>, IEquatable<IQueryFolder>
+    {
+        string Name { get; }
+        IQueryFolderCollection SubFolders { get; }
+        IQueryDefinitionCollection SavedQueries { get; }
```

Comment:
Name doesn't make sense to me: there's no reason to quality the queries returned (e.g. they can't be anything else but saved). Expected to see something like `Queries` or `QueryCollection`

Reply:
Saw that you and @MattKotsenas had a conversation about this in another file. The REST API structure is a hierarchy with limited vocabulary (e.g. a unidirected graph) and as such I expected the same types of names in the interface. That is, nodes have a name, are aware if they are a special type of node that can contain other nodes, and an enumeration of each special type of node (e.g. children). The SOAP object `QueryHierarchy` includes additional vocabulary, such as ancestor traversal, but relies on the enumerator to provide details of "children". Each child node is a descendant of `QueryItem` from what I can tell, making it less obvious than the REST API that there are children.

TL;DR - I expected names congruent with the data structure but did not find them. Great if there are "helpers" that perform certain types of common traversals automatically to keep things DRY for consumers. This new interface should be reconciled with `INode`

Reply:
will update